OneCommunityGlobal/HGNRest
# Kaikanes lessonlist routing/controller

# Description
Created routing/controllers for lesson list page so users can edit a lesson or delete a lesson in the back end. Can also fetch all lessons and a single lesson.

## Related PRS (if any):
Used the same model from 
https://github.com/OneCommunityGlobal/HGNRest/pull/635
Will eventually be used in the front end which is still in the works
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1618

## Main changes explained:
- created a temp seed function in the startup/db.js file so we can test
- created Lesson get/put/delete routes
- created LessonController that handles the logic for those routes

## How to test:
1. check into current branch
2. do `npm install` and 'npm run dev'
3. log into the app on the front end and grab your token in local storage
4. add 'Authorization' to the header with your token as the value
5. make sure 'content-type' header with 'application/json' value is in the header as well!
6. test get /put/ delete routes for Lessons 
7. get all route is localhost:4500/api/bm/lessons  (grab the _id here to use for other routes) 
8. get single/edit/delete is localhost:4500/api/bm/lesson/:id


## Screenshots or videos of changes:


https://github.com/OneCommunityGlobal/HGNRest/assets/114305309/f9005eb2-a292-4c41-8e50-8c3bee283b91

GRAB TOKEN IN APP FRONT END LOCAL STORAGE
<img width="270" alt="image" src="https://github.com/OneCommunityGlobal/HGNRest/assets/114305309/dda410e3-9308-442c-898d-810e40dc5e92">
 HEADER (I use insomnia)
![image](https://github.com/OneCommunityGlobal/HGNRest/assets/114305309/0562e588-3448-4e6a-8cf2-9ceb6e372d04)



## Note:
I just hard coded a random user and project for the seed data and will not be merging the seedDatabase function in startup/db. Its just for testing.
